src/hammer-tech/hammer_tech.py

```diff
@@ -164,6 +173,37 @@ def library_from_json(json: str) -> Library:
     return Library.from_json(json)
 
 
+# Struct that holds an extra library and possible prefix.
+class ExtraLibrary(NamedTuple('ExtraLibrary', [
+    ('prefix', Optional[PathPrefix]),
+    ('library', Library)
+])):
+    __slots__ = ()
```

Comment:
This seems like an over optimization at this point. Did you see some memory usage problem?

Reply:
I usually just put down `__slots__` by habit at this point...